code/modules/sprite_accessories/_accessory_markings.dm

```diff
@@ -1,16 +1,35 @@
 //body markings
-
 /decl/sprite_accessory/marking
 	icon = 'icons/mob/human_races/species/default_markings.dmi'
 	do_colouration = 1 //Almost all of them have it, COLOR_ADD
+	abstract_type = /decl/sprite_accessory/marking
 	//Empty list is unrestricted. Should only restrict the ones that make NO SENSE on other species,
 	//like IPC optics overlay stuff.
-	species_allowed = list(SPECIES_HUMAN)
 	var/layer_blend = ICON_OVERLAY
 	var/body_parts = list() //A list of bodyparts this covers, in organ_tag defines
 	//Reminder: BP_L_FOOT,BP_R_FOOT,BP_L_LEG,BP_R_LEG,BP_L_ARM,BP_R_ARM,BP_L_HAND,BP_R_HAND,BP_CHEST,BP_GROIN,BP_HEAD
 	var/draw_target = MARKING_TARGET_SKIN
 	var/list/disallows = list() //A list of other marking types to ban from adding when this marking is already added
+	var/list/icons = list()
+	var/mask_to_bodypart = TRUE
+
+/decl/sprite_accessory/marking/proc/get_cached_marking_icon(var/decl/bodytype/bodytype, var/bodypart, var/color = COLOR_WHITE)
+	if(!icons[bodytype])
+		icons[bodytype] = list()
+	if(!icons[bodytype][bodypart])
+		icons[bodytype][bodypart] = list()
```

Comment:
Should this be using the lazyinitlist macros to be more succinct? The repeated ifs are a bit intimidating at first.
```suggestion
	LAZYINITLIST(icons[bodytype])
	LAZYINITLIST(icons[bodytype][bodypart])
```